iree/turbine/kernel/wave/codegen/handlers.py

```diff
@@ -601,6 +640,24 @@ def handle_minimum(lhs: Value, rhs: Value, options: WaveCompileOptions) -> OpRes
         )
     return result
 
+@handle_atomic_op(atomic_min)
+def handle_atomic_min(val: Value, buffer: Value, idx: Value, options: WaveCompileOptions) -> OpResult:
+
+    value_type = val.type
+    value_element_type = get_type_or_element_type(value_type)
+    atomic_kind = None
+    if _is_float_type(value_element_type):
+        atomic_kind = arith_d.AtomicRMWKind.minimumf
+    elif _is_integer_like_type(value_element_type) and _is_signed_or_signless_type(
+        value_element_type):
+        atomic_kind = arith_d.AtomicRMWKind.mins
+
+    a = vector_d.extract(val, static_position=[0], dynamic_position=[])
```

Comment:
Could you add an assert that this is a vector of rank 1 and size 1?

Reply:
I've added check for rank and size in the handler